source/extensions/filters/network/rocketmq_proxy/active_message.cc

```diff
@@ -0,0 +1,261 @@
+#include "extensions/filters/network/rocketmq_proxy/active_message.h"
+
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/protobuf/utility.h"
+
+#include "extensions/filters/network/rocketmq_proxy/conn_manager.h"
+#include "extensions/filters/network/rocketmq_proxy/topic_route.h"
+#include "extensions/filters/network/rocketmq_proxy/well_known_names.h"
+#include "extensions/filters/network/well_known_names.h"
+
+using Envoy::Tcp::ConnectionPool::ConnectionDataPtr;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace RocketmqProxy {
+
+ActiveMessage::ActiveMessage(ConnectionManager& connectionManager, RemotingCommandPtr&& request)
+    : connection_manager_(connectionManager), request_(std::move(request)) {
+  updateActiveRequestStats();
+}
+
+ActiveMessage::~ActiveMessage() { updateActiveRequestStats(false); }
+
+void ActiveMessage::createFilterChain() { router_ = connection_manager_.config().createRouter(); }
```

Comment:
This shouldn't be about creating a specific router filter, right? I'm going to create a filter chain that contains the router filter, and maybe other filters as well

Reply:
For the moment, we only have router filter to create. Renaming it to createRouterFilter is fine, but I'd prefer to keep the chain keyword so we might add other filter in future.